File: src/main/java/net/atos/client/bag/util/JsonbConfiguration.java
/*
 * SPDX-FileCopyrightText: 2022 Atos, 2023-2024 Lifely
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.client.bag.util;

import jakarta.json.bind.Jsonb;
import jakarta.json.bind.JsonbBuilder;
import jakarta.json.bind.JsonbConfig;
import jakarta.ws.rs.ext.ContextResolver;

public class JsonbConfiguration implements ContextResolver<Jsonb> {

  private Jsonb jsonb;

  public JsonbConfiguration() {
    final JsonbConfig jsonbConfig =
        new JsonbConfig()
            .withAdapters(
                new IndicatieEnumAdapter(),
                new StatusNaamgevingEnumAdapter(),
                new StatusPandEnumAdapter(),
                new StatusWoonplaatsEnumAdapter(),
                new StatusVerblijfsobjectEnumAdapter(),
                new TypeAdresseerbaarObjectEnumAdapter(),
                new GebruiksdoelEnumAdapter(),
                new TypeOpenbareRuimteEnumAdapter());
    jsonb = JsonbBuilder.create(jsonbConfig);
  }

  @Override
  public Jsonb getContext(Class<?> type) {
    return jsonb;
  }
}

File: src/main/kotlin/nl/info/client/bag/util/JsonbConfiguration.kt
/*
 * SPDX-FileCopyrightText: 2023 Atos, 2026 INFO.nl
 * SPDX-License-Identifier: EUPL-1.2+
 */
package nl.info.client.bag.util

import jakarta.json.bind.Jsonb
import jakarta.json.bind.JsonbBuilder
import jakarta.json.bind.JsonbConfig
import jakarta.ws.rs.ext.ContextResolver

class JsonbConfiguration : ContextResolver<Jsonb> {
    private val jsonb: Jsonb = JsonbBuilder.create(
        JsonbConfig().withAdapters(
            IndicatieEnumAdapter(),
            StatusNaamgevingEnumAdapter(),
            StatusPandEnumAdapter(),
            StatusWoonplaatsEnumAdapter(),
            StatusVerblijfsobjectEnumAdapter(),
            TypeAdresseerbaarObjectEnumAdapter(),
            GebruiksdoelEnumAdapter(),
            TypeOpenbareRuimteEnumAdapter()
        )
    )

    override fun getContext(type: Class<*>): Jsonb = jsonb
}
